src/controllers/utils/modelHelper.ts

```diff
@@ -182,3 +184,52 @@ export const getProjectStats = async (
 
   return ProjectStats.parse(result[0]);
 };
+
+export const getOrCreateAttestorOrganisation = async (
+  ctx: DataHandlerContext<Store>,
+  attestor: Attestor,
+  refUID?: string,
+  attestTimestamp?: Date
+): Promise<AttestorOrganisation | undefined> => {
+  // Check if refUID is valid and not a placeholder for an empty reference
+  if (refUID && refUID !== ZERO_UID) {
+    // Attempt to find existing AttestorOrganisation
+    const attestorOrganisation = await ctx.store.get(AttestorOrganisation, {
+      where: { id: refUID.toLowerCase() },
+      relations: { organisation: true, attestor: true },
+    });
+
+    if (attestorOrganisation) {
+      ctx.log.debug(
+        `Found existing attestorOrganisation: ${attestorOrganisation}`
+      );
+      return attestorOrganisation;
+    }
+  }
+
+  // Attempt to retrieve default organisation
+  let organisation = await ctx.store.get(Organisation, ZERO_UID);
+
+  if (!organisation) {
+    ctx.log.error("No Affiliation organisation not found");
+    return undefined;
+  }
+
+  // Ensure timestamp is set
+  const timestamp = attestTimestamp || new Date();
+
+  // Generate unique key
+  const key = `NO_AFFILIATION${attestor.id}`;
+
+  const newAttestorOrganisation = new AttestorOrganisation({
+    id: key,
+    attestor,
+    organisation,
+    attestTimestamp: timestamp,
+  });
+
+  await ctx.store.upsert(newAttestorOrganisation);
```

Comment:
I think we used upsert correctly, it shouldn't make issue if an identical item has been created before.